api_app/analyzers_manager/file_analyzers/mobsf_service.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class MobSF_Service(FileAnalyzer):
+    mobsf_host: str
+    _mobsf_api_key: str
+
+    def update(self) -> bool:
+        pass
+
+    def static_analysis(self, scan_hash, headers):
+
+        scan_url = self.mobsf_host + "/api/v1/scan"
+        data = {"hash": scan_hash}
+        scan_response = requests.post(url=scan_url, data=data, headers=headers)
+        scan_response.raise_for_status()
+
+        report_url = self.mobsf_host + "/api/v1/report_json"
+        report_response = requests.post(url=report_url, data=data, headers=headers)
+        report_response.raise_for_status()
+        return report_response.json()
+
```

Comment:
about the static analysis I am worried about execution time. By default, `requests` does not set any timeout. This is potentially dangerous in IntelOwl because it could lead to jobs that never end, creating bottlenecks in the overall application.
IntelOwl cannot rely on external tools. This is the reason why we need to explicity set timeouts to all the `requests` calls that you made in this analyzer. Which value? What your experience with this tool tells you? I think that static analysis could last several seconds. 30 seconds as default could make sense?
Plus, this value should be a parameter of the analyzer and customized so the users are not bound to it.

Reply:
Though i've not used mobsf extensively but the apps i've tested did not take more than 30 seconds for static analysis. I understand it will be better if we set a parameter for this so that the end-user can modify it according to their needs.

But I wan't to understand, do we need to set this timeout because when I was implementing this analyzer, it was failing at the "dynamic analysis" stage with an error "SoftTimeLimitExceeded". So I had increased the Soft-time limit in the analyzer config.

I do understand that setting a default timeout would be better for error handling but wouldn't just configuring the soft-time limit take care of everything. Let me know what you think!

Reply:
> Even in the worst-case scenario, where our mobsf doesn't respond to the request and the request keeps hanging, wouldn't celery's Soft time limit take care of this and would end the job after specific period of time?

Yes but that would be the _last_ resort and should be treated as an unmanaged exception. It's like creating a generic Exception in python code. Ok, you get when something bad happens but you did not manage it properly cause you don't know what exactly was happening. Pretty much the same thing here. Soft time limit should be treated as the emergency exception.

About adding the timeout, we already did that for other similar services if you check other analyzers like [this](https://github.com/intelowlproject/IntelOwl/blob/master/api_app/analyzers_manager/observable_analyzers/circl_pdns.py#L39). High soft time limits and customizable timeouts, with low timeout values by default.

Regarding the values that you proposed, I think that even more than 400 could be set, maybe 1000 to handle the worst cases scenarios.

Reply:
Completely get your point about the _unmanaged exception_. I'll provide an option of request timeout parameter with a reasonable default value which can be modified as per user's need. 

I just want to understand one last thing, let's say a mobsf job has been triggered where default soft-time limit is 60 seconds but now the user has set the request timeout value as 400 seconds. Would this 400 seconds timeout be respected or the job will fail with error "SoftTimeLimitExceeded". 

Because from my limited understanding of the Intelowl, if the alloted time for a job to run is set to 60 seconds, the job will end even if we set the request timeout higher than the soft-time limit. So along with setting the request timeout, we'll have to configure the soft-time limit to a decent time period so that the job succeeds.

Let me know if my understanding is correct/incorrect?

Reply:
> I just want to understand one last thing, let's say a mobsf job has been triggered where default soft-time limit is 60 seconds but now the user has set the request timeout value as 400 seconds. Would this 400 seconds timeout be respected or the job will fail with error "SoftTimeLimitExceeded".

the job would fail with SoftTimeLimitExceeded.

> we'll have to configure the soft-time limit to a decent time period so that the job succeeds.

Exactly. In this way soft time limits exception are limited to unwanted exceptions. I understand that the timeout can be changed by the user in a way that the soft time limit could always be reached but that's something we can ignore to avoid over-engineering this